### ISSUE TITLE
Non-verbose error: Missing type annotation for S.

### ISSUE THREAD

**Issue by schumannd** [facebook/flow#6172]
when calling concat on a non-typed variable I got the below error
```
Missing type annotation for S.

     88│         this.sessions = this.sessions.concat(dataToSend);
```

It was mostly guesswork to find out what that means. Is it possible to make this error more verbose? Or is this actually an error of `flow-typed` or so?

**Comment by mrkev** [comment]
hmm, could you provide a bit more info on how to repro this? Is it at all reproducible on flow.org/try?

**Comment by brookemitchell** [comment]
I am seeing this a lot with array operations, here is an example of the change between .69 and .70

https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBTAHgBzgJwBcxCBPHDMAMQEsZCN8BBAY0JrgDswBeMAb1RhhIshQBcYAERUAkgBkAKgFEASlIA0QkcJwBDUvD0ATSQGdC+GpwDmWnQF8A3Omx4iJcpQDKhPQ14BbQQCYzNzS2sbAG0AXVRndExcAmJjDCg9AFd6MCgszjYObig6BnxffwwACm0LKslKhnswPSKuSVp6RlZ2LlQASiCdYXwMQiz8bkERkQA6Bfrm7Vm-G3CwJYw5kPwwudLu-GrKHgA+MEoAQh4+Nr7OOf1DOBMBlscXB1QgA

**Comment by brookemitchell** [comment]
I'd agree it's a duplicate of https://github.com/facebook/flow/issues/6151 , error is fixed by typing the exports return type. 

However the error isn't very descriptive and I'm a little surprised it's necessary with exports when most return types can be inferred?

Fixed example:
https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVA3AhgJzADzAF4wBtARgBoAmAXQG51hgwoBLfAUwBMwAjAJ5gALgIAObAHYBzMAIDkAZxHjOYVJ3xi4OYWGx4hALjCSArgFs+nHKVrECAOgDGcSc6zCAFKQDMtAEpGVGZWDh5+IVEJGTAcTmEzHEkVMTUNLR09KDN3YTY3MFwcLAEAMTYYYRswL2KTRWEcKWk7SjAEHW4GppaAnuaZOzAAbzB1cfjE5KKcHEd2KpsvNSIAPjA1AEIiEk6cbiDUAF8gA

**Comment by mrkev** [comment]
Closing in favor of #6151. Not inferring types at module boundaries is an optimization: it allows flow to check files in parallel without going down deep inference rabbit holes.

**Comment by brookemitchell** [comment]
That's pretty cool. Hopefully with nicer error message in the future
